FitGoal/FitGoal/Singup&Singin/CustomUIViews/AuthenticationFormView.swift

```diff
@@ -63,6 +69,31 @@ class AuthenticationFormView: UIStackView, UITextFieldDelegate {
         return true
     }
     
+    func textFieldDidBeginEditing(_ textField: UITextField) {
+        if let textField = textField as? CustomTextField {
+            if !textField.textFieldMessageLabel.isHidden {
+                textField.textFieldMessageLabel.isHidden = true
+            }
+        }
+    }
+    
+    func textFieldDidEndEditing(_ textField: UITextField) {
+        if let textField = textField as? CustomTextField {
+            guard let text = textField.text else { fatalError() }
+            delegate?.userDidEndEditing(textFieldType: textField.textFieldType, with: text)
+        }
+    }
+    
+    func showAuthenticationMessage(message: String, in textField: TextFieldType) {
+        for customTextField in self.subviews {
+            if let field = customTextField as? CustomTextField, textField == field.textFieldType {
```

Comment:
These names look weird to me, in the first `for` you call the variable `customTextField` and in the `if` you call it `field`. Shouldn't it be the other way?